staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -173,6 +173,8 @@ func (wc *watchChan) sync() error {
 	opts := []clientv3.OpOption{}
 	if wc.recursive {
 		opts = append(opts, clientv3.WithPrefix())
+		// Sort by modRevision so we send synthetic ADDED events in order of last update
+		opts = append(opts, clientv3.WithSort(clientv3.SortByModRevision, clientv3.SortAscend))
```

Comment:
Ugh, this could be really, really expensive.  Who does the sort?

Reply:
I.e. is this in the natural order in etcd or is this going to cause etcd to hold everything in memory and sort once?

Reply:
natural order is key order. etcd already holds all in memory. this would trigger an in-memory sort:

https://github.com/etcd-io/etcd/blob/24fc2a983ae39b4fe09cef3db434c812fdc97cff/etcdserver/apply.go#L313-L320